### PR TITLE
docs: add sections to tell usage of Draft PR

### PR DESCRIPTION
**Pull request by randmonkey**
**What this PR does / why we need it**:

Add a section in `CONTRIBUTING.md` to tell usage of `Draft` PRs and `do not merge` or `work in progress` labels on PRs:
- Draft PR: not ready for review
- `do not merge` label: ready for review, but not ready to merge

**Which issue this PR fixes**:

**Special notes for your reviewer**:

**PR Readiness Checklist**:

Complete these before marking the PR as `ready to review`:

- [ ] the `CHANGELOG.md` release notes have been updated to reflect any significant (and particularly user-facing) changes introduced by this PR

### CODE REVIEW

**Review by programmer04 — DISMISSED** [review]

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -29,6 +29,11 @@ If you wish to contribute code (features or bug fixes) please keep in mind the f
 - bug fix pull requests should be opened against `main` as the base branch
 - feature pull requests should be opened with `next` as the base branch
 
+If your pull request is still in progress, but you want get some early reviews before
```
Comment (pmalek) [inline CONTRIBUTING.md:32]:
```suggestion
If your pull request is still in progress, but you want to get an earl review before
```

Comment (pmalek) [reply CONTRIBUTING.md:32]: Nit: I believe this should be placed in its own section, not in `Finding work`. Something like `Working with Pull Requests` maybe.

Comment (randmonkey) [reply CONTRIBUTING.md]: updated.

Comment (pmalek) [reply CONTRIBUTING.md]: ping: https://github.com/Kong/kubernetes-ingress-controller/pull/4677#discussion_r1329856858

Comment (randmonkey) [reply CONTRIBUTING.md]: ADDED the section and a simple guide of submitting PRs.

**Review by pmalek — APPROVED** [review]
🏅

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -29,6 +29,20 @@ If you wish to contribute code (features or bug fixes) please keep in mind the f
...
+[pull request template](https://github.com/Kong/kubernetes-ingress-controller/blob/a5cfca4a97927c2878fce593929a8e48442f3127/.github/PULL_REQUEST_TEMPLATE.md). 
+A clear title and description is helpful to encourage your pull request to be reviewed and merged.
+
+If your pull request is still in progress, but you want to get early reviews before
+it is ready to be merged, please add `do not merge` or `work in progress` labels to the
+pull request. If your pull request is not ready to be reviewed yet, please convert it to `Draft`.
+`Draft` pull requests will not be reviewed until they are converted to `Open` status.
+
+
+
```
Comment (pmalek) [inline CONTRIBUTING.md:45]:
```suggestion

```

**Review by programmer04 — DISMISSED** [review]

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -29,6 +29,17 @@ If you wish to contribute code (features or bug fixes) please keep in mind the f
...
+When you finish your work and want your contributions to be included, please submit a pull request
+and include necessary information about the pull request in its description following our
+[pull request template](https://github.com/Kong/kubernetes-ingress-controller/blob/a5cfca4a97927c2878fce593929a8e48442f3127/.github/PULL_REQUEST_TEMPLATE.md). 
+A clear title and description is helpful to encourage your pull request to be reviewed and merged.
+
+If your pull request is still in progress, but you want to get early reviews before
+it is ready to be merged, please add `do not merge` or `work in progress` labels to the
+pull request. If your pull request is not ready to be reviewed yet, please convert it to `Draft`.
+`Draft` pull requests will not be reviewed until they are converted to `Open` status.
```
Comment (programmer04) [inline CONTRIBUTING.md:42]:
```suggestion
`Draft` pull requests will not be reviewed until they are converted to `Open` status.

```

**Pull request merged**